be/src/exec/pipeline/fragment_executor.cpp

```diff
@@ -234,7 +234,12 @@ Status FragmentExecutor::_prepare_runtime_state(ExecEnv* exec_env, const Unified
     int64_t query_mem_limit = _query_ctx->compute_query_mem_limit(parent_mem_tracker->limit(), per_instance_mem_limit,
                                                                   degree_of_parallelism, option_query_mem_limit);
     int64_t big_query_mem_limit = wg->use_big_query_mem_limit() ? wg->big_query_mem_limit() : -1;
-    _query_ctx->init_mem_tracker(query_mem_limit, parent_mem_tracker, big_query_mem_limit, wg.get());
+    int64_t spill_mem_limit_bytes = -1;
+    if (query_options.__isset.enable_spill && query_options.enable_spill == true) {
+        spill_mem_limit_bytes = query_mem_limit * query_options.spill_mem_limit_threshold;
```

Comment:
what if it overflow

Reply:
> what if it overflow

query_options.spill_mem_limit_threshold is never larger than 1.0, so overflow won't happen.